ts/api/backend.ts

```diff
@@ -13,13 +15,26 @@ import {
   basicResponseDecoder,
   BasicResponseType,
   composeHeaderProducers,
-  composeResponseDecoders,
   createFetchRequestForApi,
   IGetApiRequestType,
   IPostApiRequestType,
   IPutApiRequestType
 } from "../utils/api_request";
 
+//
+// Define composed types (e.g. oneOf) as they are not yet generated by
+// gen-api-models
+//
+
+// ProfileWithOrWithoutEmail is oneOf [ProfileWithEmail, ProfileWithoutEmail]
+const ProfileWithOrWithoutEmail = t.union([
+  ProfileWithEmail,
+  ProfileWithoutEmail
+]);
+
+// FullProfile is allOf [ExtendedProfile, LimitedProfile]
+const FullProfile = t.intersection([ExtendedProfile, LimitedProfile]);
```

Comment:
If I remember well LimitedProfile is a subset of ExtendedProfile so probably you don't need an intersection here